File: Programmers/lv2/top.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> heights) {
    vector<int> answer;
    int i,j;
    answer.push_back(0);
    for(i = 1 ; i < heights.size(); i ++)
    {
        bool chk = false;
        for(j = i; j >= 0; j --)
        {
            if(heights[j] > heights[i])
            {
                chk = true;
                answer.push_back(j+1);
                break;
            }
        }
        if(!chk)
            answer.push_back(0);
    }
    return answer;
}

File: Programmers/lv2/top.py
def solution(heights):
    answer = []
    answer.append(0)
    for i in range(1,len(heights)):
        chk = False
        for j in range(i,-1,-1):
            if heights[j] > heights[i] :
                chk = True
                answer.append(j+1)
                break
        if(chk == False):
            answer.append(0)
    return answer